mock/py/mockbuild/package_manager.py

```diff
@@ -16,41 +16,93 @@
 
 from . import util
 from .exception import BuildError, Error, YumError
-from .trace_decorator import traceLog
-
-def package_manager(config_opts, buildroot, plugins, bootstrap_buildroot=None):
-    pm = config_opts.get('package_manager', 'yum')
-    is_bootstrap_image = bool(buildroot.is_bootstrap and buildroot.use_bootstrap_image)
-    if pm == 'yum':
-        return Yum(config_opts, buildroot, plugins, bootstrap_buildroot, is_bootstrap_image)
-    elif pm == 'dnf':
-        if os.path.isfile(config_opts['dnf_command']) or bootstrap_buildroot is not None:
-            return Dnf(config_opts, buildroot, plugins, bootstrap_buildroot, is_bootstrap_image)
-        # RHEL without DNF and without bootstrap buildroot
-        (distribution, version) = distro.linux_distribution(full_distribution_name=False)[0:2]
-        if distribution in util.RHEL_CLONES:
-            version = int(version.split('.')[0])
-            if version < 8:
-                if ('dnf_warning' not in config_opts or config_opts['dnf_warning']) and \
-                        not config_opts['use_bootstrap_container']:
-                    print("""WARNING! WARNING! WARNING!
-You are building package for distribution which use DNF. However your system
-does not support DNF. You can continue with YUM, which will likely succeed,
-but the result may be little different.
+from .trace_decorator import traceLog, getLog
+
+fallbacks = {
+    'dnf': ['dnf', 'microdnf', 'yum'],
+    'yum': ['yum', 'dnf', 'microdnf'],
```

Comment:
MicroDNF cannot actually work without DNF. Yet. So having microdnf in fallback for yum and dnf does not have sense.  E.g. Microdnf cannot do 'builddep' command and execute dnf for that.